launchable/commands/record/case_event.py

```diff
@@ -151,6 +154,17 @@ def stderr(case: TestCase) -> str:
     def create(cls, test_path: TestPath, duration_secs: float, status,
                stdout: Optional[str] = None, stderr: Optional[str] = None,
                timestamp: Optional[str] = None, data: Optional[Dict] = None) -> Dict:
+        def _timestamp(ts: Optional[str] = None):
+            if ts is None:
+                return datetime.datetime.now(datetime.timezone.utc).isoformat()
+            try:
+                date = dateutil.parser.parse(ts)
+                if date.tzinfo is None:
+                    return date.replace(tzinfo=tzlocal()).isoformat()
```

Comment:
Question: Can we receive Time Zones except UTC at the API (Launchable) side? 
And did you confirm that the data were saved correctly?

Reply:
Yes I confirmed the following XML were accepted correctly.

```
<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="tests.test_version.VersionTest" tests="1" file="tests/test_version.py" time="0.003" timestamp="2024-04-24T16:07:51" failures="0" errors="0" skipped="0">
	<testcase classname="tests.test_version.VersionTest" name="test_version" time="0.003" timestamp="2024-04-24T16:07:51" file="tests/test_version.py" line="10"/>
</testsuite>
```

```
$ launchable --log-level=audit  record tests  file test-results/TEST-tests.test_version.VersionTest.xml 
AUDIT:launchable:send request method:get path:http://localhost:8080/intake/organizations/launchableinc-2/workspaces/test/builds/16 headers:{'User-Agent': 'Launchable/1.87.1.dev1+g051949b (Python 3.11.4, macOS-13.3-arm64-arm-64bit) TestRunner/file', 'Content-Type': 'application/json', 'Authorization': 'Bearer v1:launchableinc-2/test:368deb306ea044c88e369f96fddb9a68'} args:None
AUDIT:launchable:send request method:get path:http://localhost:8080/intake/organizations/launchableinc-2/workspaces/test/slack/notification/key/list headers:{'User-Agent': 'Launchable/1.87.1.dev1+g051949b (Python 3.11.4, macOS-13.3-arm64-arm-64bit) TestRunner/file', 'Content-Type': 'application/json', 'Authorization': 'Bearer v1:launchableinc-2/test:368deb306ea044c88e369f96fddb9a68'} args:None
AUDIT:launchable:send request method:post path:http://localhost:8080/intake/organizations/launchableinc-2/workspaces/test/builds/16/test_sessions/32/events headers:{'User-Agent': 'Launchable/1.87.1.dev1+g051949b (Python 3.11.4, macOS-13.3-arm64-arm-64bit) TestRunner/file', 'Content-Type': 'application/json', 'Content-Encoding': 'gzip', 'Authorization': 'Bearer v1:launchableinc-2/test:368deb306ea044c88e369f96fddb9a68'} args:{'events': [{'type': 'case', 'testPath': [{'type': 'file', 'name': 'tests/test_version.py'}, {'type': 'testsuite', 'name': 'tests.test_version.VersionTest'}, {'type': 'testcase', 'name': 'test_version'}], 'duration': 0.003, 'status': 1, 'stdout': '', 'stderr': '', 'createdAt': '2024-04-24T16:07:51+09:00', 'data': {'lineNumber': 10}}], 'testRunner': 'file', 'group': '', 'noBuild': False, 'metadata': {}}
Launchable recorded tests for build 16 (test session 32) to workspace launchableinc-2/test from 1 files:

|   Files found |   Tests found |   Tests passed |   Tests failed |   Total duration (min) |
|---------------|---------------|----------------|----------------|------------------------|
|             1 |             1 |              1 |              0 |                   0.00 |

Visit https://app.launchableinc.com/organizations/launchableinc-2/workspaces/test/test-sessions/32 to view uploaded test results (or run `launchable inspect tests --test-session-id 32`)
```

Reply:
> And did you confirm that the data were saved correctly?

Yes, when checking DB, it was stored as "created_at": "2024-04-24T07:07:51Z".

Reply:
>> 'createdAt': '2024-04-24T16:07:51+09:00', 'data': {'lineNumber': 10}}]
>  it was stored as "created_at": "2024-04-24T07:07:51Z".

Nice 👍